core/types/transaction.go

```diff
@@ -508,6 +508,7 @@ func (s *TxByPriceAndTime) Pop() interface{} {
 	old := *s
 	n := len(old)
 	x := old[n-1]
+	old[n-1] = nil
 	*s = old[0 : n-1]
```

Comment:
Interesting idea -- but I think this should not be needed. Instead of nilling the item, we resize the slice. 

Taken to the extreme, imagine we have a slice with 100 pointers, and then do `myslice = myslice[:0]` -- that should free up all the pointers for the gc, without us explicitly nilling them. It's the same case here.

Reply:
I don't think resizing would help since the same underlying array is referenced.

This is the example for a priority queue from the standard lib:

```
func (pq *PriorityQueue) Pop() any {
	old := *pq
	n := len(old)
	item := old[n-1]
	old[n-1] = nil  // avoid memory leak
	item.index = -1 // for safety
	*pq = old[0 : n-1]
	return item
}
```
https://pkg.go.dev/container/heap